src/smolagents/default_tools.py

```diff
@@ -569,6 +569,28 @@ def decode(self, outputs):
         return self.pre_processor.batch_decode(outputs, skip_special_tokens=True)[0]
 
 
+class ManagedAgentTool(Tool):
+    inputs = {
+        "task": {"type": "string", "description": "a very detailed description of the task the agent needs to perform"}
+    }
+    output_type = "string"
+
+    def __init__(self, agent=None):
+        if agent is None:
+            raise ValueError("ManagedAgentTool requires an agent to delegate tasks to.")
+        self.agent = agent
+        self.name = getattr(self.agent, "name", self.agent.__class__.__name__)
+        self.description = getattr(
+            self.agent,
+            "description",
+            f"Delegates a task to the '{self.agent.name}' agent who is responsible for: {self.agent.description}.",
```

Comment:
You are accessing `self.agent.description` if "description" attribute does not exist in `self.agent`!